tofd/pars/R3BTofDCal2HitPar.cxx

```diff
@@ -297,1099 +352,2668 @@ void R3BTofDCal2HitPar::Exec(Option_t* option)
+				    //LOG(info) << "Plane  " << i + 1 << " Bar " << j + 1 << " max  " << max;
+				    //LOG(info) << "Plane  " << i + 1 << " Bar " << j + 1 << " veff  " << veff;
+				    par->SetZoffset(Zoffset);
+			    }
+		    }
+	    }
+	    fHitPar->setChanged();
     }
+*/
 
-    if (fParameter == 4)
-    {
-        // Z correction for each plane
-        LOG(warn) << "Calling function zcorr";
-        Double_t para[8];
-        Double_t pars[3];
-        pars[0] = 0.;
-        pars[1] = 0.;
-        pars[2] = 0.;
-        Int_t min = 0, max = fMaxQ; // select range for peak search
-        for (Int_t i = 0; i < fNofPlanes; i++)
-        {
-            for (Int_t j = 0; j < fPaddlesPerPlane; j++)
-            {
-                if (fhQvsPos[i][j])
-                {
-                    auto par = fHitPar->GetModuleParAt(i + 1, j + 1);
-                    std::cout << "Calling Plane: " << i + 1 << " Bar " << j + 1 << "\n";
-                    Int_t index = i * fPaddlesPerPlane + j;
-                    zcorr(fhQvsPos[i][j], min, max, pars, index);
-                    std::cout << "Write parameter: " << pars[0] << " " << pars[1] << " " << pars[2] << "\n";
-                    par->SetPar1za(pars[0]);
-                    par->SetPar1zb(pars[1]);
-                    par->SetPar1zc(pars[2]);
-                }
-            }
-        }
-        fHitPar->setChanged();
-    }
 
-    for (Int_t i = 0; i < N_TOFD_HIT_PLANE_MAX; i++)
-    {
-        if (fh_tofd_TotPm[i])
-            fh_tofd_TotPm[i]->Write(); // control histogram for ToT
-        if (fhTsync[i])
-            fhTsync[i]->Write(); // histogram for sync calculation
-        if (fhTdiff[i])
-            fhTdiff[i]->Write(); // histogram for offset and veff calculation
-        for (Int_t j = 0; j < N_TOFD_HIT_PADDLE_MAX; j++)
-        {
-            if (fh1_tofsync[i][j])
-                fh1_tofsync[i][j]->Write(); // histogram for ToF sync calculation
-            if (fhLogTot1vsLogTot2[i][j])
-                fhLogTot1vsLogTot2[i][j]->Write(); // control histogram Log(ToT) Pm1 vs Log(ToT) Pm2
-            if (fhSqrtQvsPosToT[i][j])
-                fhSqrtQvsPosToT[i][j]->Write(); // histogram for ToT offset calculation
-            if (fhQvsPos[i][j])
-                fhQvsPos[i][j]->Write(); // histogram for charge fit
-            if (!fTofdSmiley)
-            {
-                if (fhTot1vsPos[i][j])
-                    fhTot1vsPos[i][j]->Write(); // histogram for position dependence of charge 1
-                if (fhTot2vsPos[i][j])
-                    fhTot2vsPos[i][j]->Write(); // histogram for position dependence of charge 2
-            }
-            /*
-            if (fhTot1vsTot2[i][j])
-                fhTot1vsTot2[i][j]->Write(); // control histogram ToT Pm1 vs ToT Pm2
-            */
-        }
-    }
+void R3BTofDCal2HitPar::FinishTask()
+{	
+
+	if (fParameter == 10)
+	{
+		// Determine time offset of the 2 PMTs of one paddle. This procedure
```

Comment:
It is only difference between fParameter==10 and fParameter==1 the difference in time between the up/down PMT and writing the histograms? Does this need a separate step?